terraform/eks/iam.tf

```diff
@@ -0,0 +1,186 @@
+#---
+# Provides a role for the cluster-autoscaler
+#
+# Note that the resource spec will result in a failure if we have
+# multiple ASGs. I would like to find a way to successfully pass
+# in a list.
+#---
+
+resource "aws_iam_role" "cluster_autoscaler" {
+  name = "cluster-autoscaler-${var.environment}-${var.region}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "AWS": "${module.eks.worker_iam_role_arn}"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "cluster_autoscaler" {
+  name = "cluster-autoscaler-${var.environment}-${var.region}"
+  role = "${aws_iam_role.cluster_autoscaler.name}"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "autoscaling:DescribeAutoScalingGroups",
+                "autoscaling:DescribeAutoScalingInstances",
+                "autoscaling:DescribeTags",
+                "autoscaling:SetDesiredCapacity",
+                "autoscaling:TerminateInstanceInAutoScalingGroup",
+                "autoscaling:DescribeLaunchConfigurations"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "worker-policy" {
```

Comment:
I would consider using the `aws_iam_policy_document` data provider for the actual policy vs having it inline. See: https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html